{"@class": "de.fau.cs.inf2.cthree.data.Cluster", "id": "0a084ff33d60748f98305f39037d2fc5", "detectedBy": ["DIFF_DBSCAN", "AST_HIERARCHICAL", "AST_DBSCAN"], "members": [{"@class": "de.fau.cs.inf2.cthree.data.CodeChange", "id": "518a51ccc15a529815e3d90c77f453b5", "repository": "http://git.code.sf.net/p/drjava/git_repo", "fileName": "drjava/src/edu/rice/cs/drjava/model/GlobalModelJUnitTest.java", "commitBeforeChange": "771a0aee635c62bc6d0a2000fe605dc1113ef2b9", "commitAfterChange": "d1e56ac5c281c4b03c498f59138895f0f26350fa", "methodNumberBeforeChange": 3, "methodNumberAfterChange": 3, "signatureBeforeChange": "   public void xtestElspethOneJUnitError() throws Exception", "signatureAfterChange": "   public void testElspethOneJUnitError() throws Exception", "diff": ["-   */", "-  public void xtestElspethOneJUnitError() throws Exception {", "+  /** Tests that a JUnit file with an error is reported to have an error. */", "+  public void testElspethOneJUnitError() throws Exception {", "-    synchronized(listener) {", "-      doc.startJUnit();", "-      if (printMessages) System.out.println(\"waiting for test\");", "-      listener.wait();", "-    }", "-    if (printMessages) System.out.println(\"after test\");", "+    ", "+    _runJUnit(doc);"]}, {"@class": "de.fau.cs.inf2.cthree.data.CodeChange", "id": "8a9ad125d28aa9bc1269e7997290c52a", "repository": "http://git.code.sf.net/p/drjava/git_repo", "fileName": "drjava/src/edu/rice/cs/drjava/model/GlobalModelJUnitTest.java", "commitBeforeChange": "771a0aee635c62bc6d0a2000fe605dc1113ef2b9", "commitAfterChange": "d1e56ac5c281c4b03c498f59138895f0f26350fa", "methodNumberBeforeChange": 4, "methodNumberAfterChange": 4, "signatureBeforeChange": "   public void xtestRealError() throws Exception", "signatureAfterChange": "   public void testRealError() throws Exception", "diff": ["-   */", "-  public void xtestRealError() throws Exception {", "+  /** Tests that a test class which throws a *real* Error (not an Exception) is handled correctly. */", "+  public void testRealError() throws Exception {", "-    synchronized(listener) {", "-      doc.startJUnit();", "-      if (printMessages) System.out.println(\"waiting for test\");", "-      listener.wait();", "-    }", "-    if (printMessages) System.out.println(\"after test\");", "+    ", "+    _runJUnit(doc);"]}], "num": 17479}